### PR TITLE
feat(ingest/superset): add domains and platform_instance support

### PR DESCRIPTION
**Pull request by rinzool**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Description
Hi, in my company we really need a simple way to add a domain to chart/dashboard entities coming from Superset.
So I copied what is done for [Glue ingestion](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/aws/glue.py#L136) and add a `domain` argument in the config

I tried locally and it works as I expected :ok_hand: 

Furthermore I add a `platform_instance`  argument and deprecate `env` argument that commit https://github.com/datahub-project/datahub/pull/9647/commits/b0885122d2d994a4fc81e0b09e1f8821a77a9f46 (It may be done in another PR if necessary)

Example output when running ingestion:

Config:
```yaml
source:
  type: superset
  config:
    # Coordinates
    connect_uri: ***
    env: "STG"
    platform_instance: "my_awesome_platform"
    domain: 
      my-domain:
        allow: [".*"]

    # Credentials
    username: admin
    password: ***
    provider: db
```

Output:
```
ConfigurationWarning: env is deprecated and will be removed in a future release. Please use platform_instance instead.
...
{'events_produced': 455,
 'events_produced_per_sec': 44,
 'entities': {'dashboard': ['urn:li:dashboard:(superset,my_awesome_platform.623)',
                            'urn:li:dashboard:(superset,my_awesome_platform.433)',
                            'urn:li:dashboard:(superset,my_awesome_platform.171)',
                            'urn:li:dashboard:(superset,my_awesome_platform.476)',
                            'urn:li:dashboard:(superset,my_awesome_platform.385)',
                            'urn:li:dashboard:(superset,my_awesome_platform.239)',
                            'urn:li:dashboard:(superset,my_awesome_platform.219)',
                            'urn:li:dashboard:(superset,my_awesome_platform.188)',
                            'urn:li:dashboard:(superset,my_awesome_platform.155)',
                            'urn:li:dashboard:(superset,my_awesome_platform.19)',
```

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 1]
```diff
@@ -235,7 +247,7 @@ def get_datasource_urn_from_id(self, datasource_id):
         return None
 
     def construct_dashboard_from_api_data(self, dashboard_data):
-        dashboard_urn = f"urn:li:dashboard:({self.platform},{dashboard_data['id']})"
+        dashboard_urn = f"urn:li:dashboard:({self.platform},{self.platform_instance}{dashboard_data['id']})"
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:250]: let's use our make_chart_urn and make_dashboard_urn helpers - they also have a proper platform_instance kwargs, which should also let us avoid needing `self.platform_instance`

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 2]
```diff
@@ -419,3 +441,18 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
 
     def get_report(self) -> StaleEntityRemovalSourceReport:
         return self.report
+
+    def _gen_domain_urn(self, title: str) -> Optional[str]:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:445]: we have a "DomainRegistry" helper, which lets us convert domain names into urns using a lookup against DataHub - let's see if we can reuse the same pattern

A good example to copy would be the kafka integration's domain support: https://github.com/datahub-project/datahub/blob/1741c07d769f56a9cf066172725384b4e8780839/metadata-ingestion/src/datahub/ingestion/source/kafka.py#L445

File: metadata-ingestion/src/datahub/ingestion/source/superset.py (hunk 1 above)
Comment (rinzool) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py]: Thank you for the tip :+1: 
It works fine perfectly :ok_hand: 


Example of logs:
```
 'entities': {'dashboard': ['urn:li:dashboard:(superset,prod-bedrock.508)',
                            'urn:li:dashboard:(superset,prod-bedrock.543)',
                            'urn:li:dashboard:(superset,prod-bedrock.242)',
                            'urn:li:dashboard:(superset,prod-bedrock.337)',
                            'urn:li:dashboard:(superset,prod-bedrock.509)',
                            'urn:li:dashboard:(superset,prod-bedrock.567)',
                            'urn:li:dashboard:(superset,prod-bedrock.118)',
                            'urn:li:dashboard:(superset,prod-bedrock.428)',
                            'urn:li:dashboard:(superset,prod-bedrock.340)',
                            'urn:li:dashboard:(superset,prod-bedrock.183)',
                            '... sampled of 212 total elements'],
              'chart': ['urn:li:chart:(superset,prod-bedrock.3143)',
                        'urn:li:chart:(superset,prod-bedrock.9784)',
                        'urn:li:chart:(superset,prod-bedrock.11400)',
                        'urn:li:chart:(superset,prod-bedrock.11399)',
                        'urn:li:chart:(superset,prod-bedrock.11398)',
                        'urn:li:chart:(superset,prod-bedrock.10060)',
                        'urn:li:chart:(superset,prod-bedrock.10052)',
                        'urn:li:chart:(superset,prod-bedrock.10050)',
                        'urn:li:chart:(superset,prod-bedrock.10046)',
                        'urn:li:chart:(superset,prod-bedrock.11396)',
```

File: metadata-ingestion/src/datahub/ingestion/source/superset.py (hunk 2 above)
Comment (rinzool) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:462]: Ok thanks, it's done :+1: Let me know if it is what you expected

Comment (rinzool) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:462]: Although, I removed the `_get_domain_urn` to have only one function `_get_domain_wu` (I don't think it's relevant to have two distinct functions here)

**Review by hsheth2 — APPROVED** [review]
LGTM

**Comment by anshbansal** [comment]
@rinzool We merged in your other superset PRs. Seems they caused a conflict. If you could resolve that and ensure lint is passing we can get this merged in

**Comment by rinzool** [comment]
> @rinzool We merged in your other superset PRs. Seems they caused a conflict. If you could resolve that and ensure lint is passing we can get this merged in

Yep with 3 PRs I expected some conflicts :sweat_smile: I resolved them, CI should be successful (it's ok locally)

**Pull request merged**